stdlib/public/core/String.swift

```diff
@@ -44,6 +44,12 @@ public protocol StringProtocol
   func lowercased() -> String
   func uppercased() -> String
 
+  /// Creates a string having the given substring's unicode scalars as its content.
+  ///
+  /// - Parameters:
+  ///   - content: The `unicodeScalars` of a subsequence of this string.
```

Comment:
I'd recommend matching the documentation in concrete implementations of this method for consistent capitalization and style. Also consider if this method should have a documented complexity.

Reply:
Found the concrete implementations here:

https://github.com/apple/swift/blob/37f7493c899864170594b16c550f853c547e5b0c/stdlib/public/core/Substring.swift.gyb#L476

https://github.com/apple/swift/blob/37f7493c899864170594b16c550f853c547e5b0c/stdlib/public/core/Substring.swift.gyb#L508

The complexities of these methods are different. In the case of a `Substring` it is O(1) and there is no copying of the contents, only slicing. For `String`, it is O(n) and a full copy is performed. Should the protocol documentation note this distinction, give the upper bound, or not include a documented complexity?